dashboard/app/controllers/api/v1/pd/workshops_controller.rb

```diff
@@ -39,11 +39,17 @@ def workshops_user_enrolled_in
     enrollments = ::Pd::Enrollment.for_user(current_user).all.reject do |enrollment|
       enrollment.workshop&.future_or_current_teachercon_or_fit?
     end
-    workshops = enrollments.map do |enrollment|
-      Api::V1::Pd::WorkshopSerializer.new(enrollment.workshop, scope: {enrollment_code: enrollment.try(:code)}).attributes
+
+    # Adds information about whether the user actually attended the workshop in question
```

Comment:
Feels a bit janky to be adding information about a user's attendance to a method in the workshop serializer, but this method is only used in this one place as far as I can tell. 

I added the user's ID to the serialized data as well, not because I actually used it but because when looking at the object in the browser it was confusing that all of the information was about the workshop itself (except `attended`). I thought adding the user ID might help demystify that a little bit, but can remove.

Reply:
I'd omit the `user_id` here; instead, the serializer already includes an `enrollment_code` which is specific to one attendee, so it's not that strange to add another bit of attendee-specific information.

However, I'd recommend against adding `attended` here in the controller, and instead modify [the serializer itself](https://github.com/code-dot-org/code-dot-org/blob/7ce39ea44404fab7d3648afe6917e18b4861a118/dashboard/app/serializers/api/v1/pd/workshop_serializer.rb#L31) to do this itself when an `enrollment_code` is in scope.  I'd be happy to take a look at this together.

You can read more about [`ActiveModel::Serializer` here](https://github.com/rails-api/active_model_serializers/blob/0-10-stable/docs/general/getting_started.md).

Reply:
I understand what you're saying, but just took a look at this and the best approach isn't clear to me. Would love to pair when you have some time.

Reply:
Discussed offline and resolved.